test/read-throttling.sh

```diff
@@ -51,28 +80,35 @@ trap "cleanup" EXIT
 fpod=$( get_running_pod fluentd )
 
 # generate throttle config with invalid YAML
-os::log::debug "$( oc label node --all logging-infra-fluentd- 2>&1 || : )"
-os::cmd::try_until_text "oc get daemonset logging-fluentd -o jsonpath='{ .status.numberReady }'" "0" $FLUENTD_WAIT_TIME
+stop_fluentd
 os::log::debug "$( oc patch configmap/logging-fluentd --type=json \
    --patch '[{ "op": "replace", "path": "/data/throttle-config.yaml", "value": "\
     test-proj: read_lines_limit: bogus-value"}]' )"
-os::log::debug "$( oc label node --all logging-infra-fluentd=true 2>&1 || : )"
-os::cmd::try_until_text "oc get pods -l component=fluentd" "^logging-fluentd-.* Running "
+start_fluentd
 fpod=$( get_running_pod fluentd )
 # should have fluentd log messages like this
 os::cmd::expect_success_and_text "oc logs $fpod" "Could not parse YAML file"
 
+# generate a throttle config that properly generates different pos files
+stop_fluentd
+os::log::debug "$( oc patch configmap/logging-fluentd --type=json \
+   --patch '[{ "op": "replace", "path": "/data/throttle-config.yaml", "value": "\
+    test-proj:\n  read_lines_limit: 5\n.operations:\n  read_lines_limit: 5"}]' )"
+start_fluentd
+check_fluentd_pod_for_files '/var/log/es-container-test-proj.log.pos' '/var/log/es-container-openshift-operations.log.pos'
+check_fluentd_pod_file_content_for '/var/log/es-container-openshift-operations.log.pos' '[^ ]*'
```

Comment:
You're trying to see if the file contains zero or more non-space characters?  Won't that always succeed?  Is there something more concrete that you could search the file for to see if it correctly constructed a pos file?

Reply:
Likely... I had thought that the file should contain something if its actually reading in log files. I could specifically check for specific project entries like `_default_`